src/common/post.ts

```diff
@@ -250,17 +250,32 @@ export type CreateSourcePostModeration = Omit<
   > & {
     contentHtml?: string;
     externalLink?: string;
+    postId?: string;
   };
 
 export const createSourcePostModeration = async (
   con: DataSource | EntityManager,
   args: CreateSourcePostModeration,
 ) => {
-  const newPost = con.getRepository(SourcePostModeration).create({
+  if (args.postId) {
+    const post = await con
+      .getRepository(Post)
+      .findOneByOrFail({ id: args.postId });
+
+    const isUnauthorizedEdit =
+      (post.type === PostType.Freeform && args.createdById !== post.authorId) ||
+      (post.type !== PostType.Freeform && args.createdById !== post.scoutId);
+
+    if (isUnauthorizedEdit) {
+      throw new ForbiddenError('Cannot edit posts created by other users');
+    }
+  }
+
+  const newModerationEntry = con.getRepository(SourcePostModeration).create({
     ...args,
     status: SourcePostModerationStatus.Pending,
   });
-  return await con.getRepository(SourcePostModeration).save(newPost);
+  return await con.getRepository(SourcePostModeration).save(newModerationEntry);
```

Comment:
It will create a new row in SourcePostModeration right?

Reply:
Correct!